mmseg/datasets/transforms/transforms.py

```diff
@@ -1226,3 +1226,199 @@ def __repr__(self):
+
+    Modified Keys:
+
+    - img
+
+    Args:
+        mean (float, optional): the mean to subtract by
+            Defaults to None.
+        std (float, optional): the standard deviation to divide by
+            Defaults to None.
+    """
+
+    def __init__(self,
+                 mean: Optional[Union[float, Iterable[float]]] = None,
+                 std: Optional[Union[float, Iterable[float]]] = None,
+                 channel_wise: bool = False) -> None:
+        self.mean = mean
+        self.std = std
+        self.channel_wise = channel_wise
+
+    def _normalize(self, img: np.ndarray, mean=None, std=None):
+        slices = np.ones_like(img, dtype=bool)
+        if not slices.any():
+            return img
+
+        _mean = mean if mean is not None else np.mean(img[slices])
+        _std = std if std is not None else np.std(img[slices])
+
+        if np.isscalar(_std):
+            if _std == 0.0:
+                _std = 1.0
+        else:
+            _std = _std[slices]
+            _std[_std == 0.0] = 1.0
+
+        img[slices] = (img[slices] - _mean) / _std
+        return img
+
+    def znorm(self, img):
+        if self.channel_wise:
+            if self.mean is not None and len(self.mean) != len(img):
+                err_str = (f'img has {len(img)} channels, '
+                           f'but mean has {len(self.mean)}.')
+                raise ValueError(err_str)
+            if self.std is not None and len(self.std) != len(img):
+                err_str = (f'img has {len(img)} channels, '
+                           f'but std has {len(self.std)}.')
+                raise ValueError(err_str)
+
+            for i, d in enumerate(img):
+                img[i] = self._normalize(
+                    d,
+                    mean=self.mean[i] if self.mean is not None else None,
+                    std=self.std[i] if self.std is not None else None,
+                )
+        else:
+            img = self._normalize(img, self.mean, self.std)
+
+        return img
+
+    def transform(self, results: dict) -> dict:
+        img = results['img']
+        img = self.znorm(img)
+        results['img'] = img
+        return results
+
+    def __repr__(self):
+        repr_str = self.__class__.__name__
+        repr_str += f'(mean={self.mean}, '
+        repr_str += f'std={self.std}, '
+        repr_str += f'channel_wise={self.channel_wise})'
+        return repr_str
+
+
+@TRANSFORMS.register_module()
+class ClampIntensity(BaseTransform):
+    """clamp intensity.
+
+    Required Keys:
+
+    - img
+
+    Modified Keys:
+
+    - img
+
+    Args:
+        min (float, optional): Minimum target intensity
+        max (float, optional): Maximum target intensity
+    """
+
+    def __init__(self,
+                 t_min: Optional[float] = None,
+                 t_max: Optional[float] = None) -> None:
+        self.t_min = t_min
+        self.t_max = t_max
+
+    def clamp(self, img):
+        return np.clip(img, self.t_min, self.t_max)
```

Comment:
Perhaps we can delete L1411-1412 since `np.clip()` could be directly used in `def transform()`?